sycl/include/syclcompat/math.hpp

```diff
@@ -177,6 +190,36 @@ inline constexpr RetT extend_vbinary2(AT a, BT b, RetT c,
   return sycl::vec<Tint, 2>{temp[0], temp[1]}.template as<sycl::vec<RetT, 1>>();
 }
 
+template <typename RetT, bool NeedSat, bool NeedAdd, typename AT, typename BT,
+          typename BinaryOperation>
+inline constexpr RetT extend_vbinary4(AT a, BT b, RetT c,
+                                      BinaryOperation binary_op) {
+  static_assert(std::is_integral_v<AT> && std::is_integral_v<BT> &&
+                std::is_integral_v<RetT> && sizeof(AT) == 4 &&
+                sizeof(BT) == 4 && sizeof(RetT) == 4);
+  sycl::vec<int16_t, 4> extend_a = extractAndExtend4(a);
+  sycl::vec<int16_t, 4> extend_b = extractAndExtend4(b);
+  sycl::vec<int16_t, 4> temp{
+      binary_op(extend_a[0], extend_b[0]), binary_op(extend_a[1], extend_b[1]),
+      binary_op(extend_a[2], extend_b[2]), binary_op(extend_a[3], extend_b[3])};
+  using Tint =
+      typename std::conditional<std::is_signed_v<RetT>, int8_t, uint8_t>::type;
+
+  if constexpr (NeedSat) {
+    int16_t min_val = 0, max_val = 0;
+    min_val = std::numeric_limits<Tint>::min();
+    max_val = std::numeric_limits<Tint>::max();
+    temp = detail::clamp(temp, {min_val, min_val, min_val, min_val},
```

Comment:
We can use the following code to construct a 4 element `min_val`/`max_val` vector.
https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_vec_interface
`explicit constexpr vec(const DataT& arg);`

```
sycl::vec<int16_t, 4>(min_val), sycl::vec<int16_t, 4>(max_val)
```

Reply:
Will apply for the v2 variant as well (just to keep it consistent, as the v2 extend got merged already).